File: backend/analytics/src/main/java/com/energy/analytics/dto/rest/CountryFlowSummaryDTO.java
package com.energy.analytics.dto.rest;

import java.time.Instant;

public record CountryFlowSummaryDTO(
   Instant timestamp,
   String country,
   float totalImportMW,
   float totalExportMW,
   float netFlow,
   double importShortTermChangePercentage,
   double exportShortTermChangePercentage,
   double importLongTermChangePercentage,
   double exportLongTermChangePercentage
) {}

File: frontend/src/app/core/model/dto/country-flow-summary.dto.ts
export interface CrossborderFlowSummaryDTO {
  timestamp: string;
  country: string;
  totalImportMW: number;
  totalExportMW: number;
  netFlow: number;
  importShortTermChangePercentage: number;
  exportShortTermChangePercentage: number;
  importLongTermChangePercentage: number;
  exportLongTermChangePercentage: number;
}
